smarts/core/agent.py

```diff
@@ -146,7 +146,27 @@ def __post_init__(self):
     def replace(self, **kwargs) -> "AgentSpec":
         """Return a copy of this AgentSpec with the given fields updated."""
 
-        return replace(self, **kwargs)
+        replacements = [
+            ("policy_builder", "agent_builder"),
+            ("policy_params", "agent_params"),
+            ("perform_self_test", None),
```

Comment:
Would `None` be informative enough for the warning message below?

Reply:
I used `None` because it should not be in the `kwargs`. I can fix this I think by asserting that `None` is not in the kwargs.

Reply:
Maybe I wasn't being clear due to lack of context, what should the users do if they use "perform_self_test"? Should they get a deprecated warning that instructs them not to use this, because there's no replacement (`None`)?

Reply:
I see what you mean now, I have addressed it.